### PR TITLE
Priority queue cleanup

### PR DESCRIPTION
**Pull request by jlhonora**
- Usage of defer for some mutex unlocks
- Use disposeLock to check if queue is disposed

### CODE REVIEW

File: queue/priority_queue.go [hunk 1]
```diff
@@ -120,8 +120,9 @@ func (pq *PriorityQueue) Put(items ...Item) error {
 	}
 
 	pq.lock.Lock()
+	defer pq.lock.Unlock()
```
Comment (alexandercampbell-wf) [inline queue/priority_queue.go]: In our experience, defer has a performance overhead. Have you re-run the benchmarks on `Put` to see how they are affected?

Comment (jlhonora) [inline queue/priority_queue.go]: Good catch. I changed that part of the code just to be consistent with the rest of the functions.

Anyways, how can you test it consistently? I just ran two benchmarks, one was faster and one was slower than the current version:

```
workiva/go-datastructures > go test -bench .
PASS
BenchmarkPriorityQueue	 1000000	      2272 ns/op
BenchmarkQueue	 5000000	       388 ns/op
BenchmarkChannel	 1000000	      1129 ns/op
BenchmarkQueuePut	    5000	    243654 ns/op
BenchmarkQueueGet	   10000	    211298 ns/op
BenchmarkQueuePoll	   10000	    178354 ns/op
BenchmarkExecuteInParallel	   10000	    105245 ns/op
BenchmarkRBLifeCycle	  300000	      5260 ns/op
BenchmarkRBPut	10000000	       749 ns/op
BenchmarkRBGet	10000000	       193 ns/op
ok  	github.com/workiva/go-datastructures/queue	76.658s
```

```
jlhonora/go-datastructures > go test -bench .
PASS
BenchmarkPriorityQueue	 1000000	      2527 ns/op
BenchmarkQueue	 5000000	       386 ns/op
BenchmarkChannel	 1000000	      1147 ns/op
BenchmarkQueuePut	    5000	    247075 ns/op
BenchmarkQueueGet	   10000	    329593 ns/op
BenchmarkQueuePoll	   10000	    231860 ns/op
BenchmarkExecuteInParallel	   10000	    108442 ns/op
BenchmarkRBLifeCycle	  300000	      5350 ns/op
BenchmarkRBPut	 5000000	       230 ns/op
BenchmarkRBGet	10000000	       644 ns/op
ok  	github.com/jlhonora/go-datastructures/queue	59.819s
jlhonora/go-datastructures > go test -bench .
PASS
BenchmarkPriorityQueue	 1000000	      2350 ns/op
BenchmarkQueue	 5000000	       373 ns/op
BenchmarkChannel	 1000000	      1136 ns/op
BenchmarkQueuePut	    5000	    239267 ns/op
BenchmarkQueueGet	   10000	    191165 ns/op
BenchmarkQueuePoll	   10000	    181181 ns/op
BenchmarkExecuteInParallel	   10000	    152712 ns/op
BenchmarkRBLifeCycle	  300000	      7282 ns/op
BenchmarkRBPut	 5000000	       219 ns/op
BenchmarkRBGet	10000000	       511 ns/op
ok  	github.com/jlhonora/go-datastructures/queue	54.827s
```

Do you have a more standard way to run the benchmark suite?

Comment (alexandercampbell-wf) [inline queue/priority_queue.go]: I ran the benchmarks locally and got a similar result to what your benchmarks show. The defer version is actually faster for Put and Get. Interesting.

4-core i7 2.8 GHz, 16GB memory

```
[master][~/workspace/src/github.com/Workiva/go-datastructures/queue]$ go test -bench .
PASS
BenchmarkPriorityQueue-8         1000000              1618 ns/op
BenchmarkQueue-8         3000000               524 ns/op
BenchmarkChannel-8        500000              3726 ns/op
BenchmarkQueuePut-8        20000             84385 ns/op
BenchmarkQueueGet-8        20000             84042 ns/op
BenchmarkQueuePoll-8       20000             76694 ns/op
BenchmarkExecuteInParallel-8       30000             52584 ns/op
BenchmarkRBLifeCycle-8  20000000               111 ns/op
BenchmarkRBPut-8        20000000               112 ns/op
BenchmarkRBGet-8        30000000               247 ns/op
ok      github.com/Workiva/go-datastructures/queue      62.393s
[master][~/workspace/src/github.com/Workiva/go-datastructures/queue]$ gco jlhorona/master
Note: checking out 'jlhorona/master'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a942133... Priority queue cleanup
[a942133][~/workspace/src/github.com/Workiva/go-datastructures/queue]$ go test -bench .
PASS
BenchmarkPriorityQueue-8         1000000              1621 ns/op
BenchmarkQueue-8         3000000               503 ns/op
BenchmarkChannel-8        500000              3532 ns/op
BenchmarkQueuePut-8        20000             79595 ns/op
BenchmarkQueueGet-8        20000             81775 ns/op
BenchmarkQueuePoll-8       20000             78193 ns/op
BenchmarkExecuteInParallel-8       30000             60249 ns/op
BenchmarkRBLifeCycle-8  20000000               101 ns/op
BenchmarkRBPut-8        20000000                83.9 ns/op
BenchmarkRBGet-8        30000000               207 ns/op
ok      github.com/Workiva/go-datastructures/queue      59.569s
[a942133][~/workspace/src/github.com/Workiva/go-datastructures/queue]$
```

**Comment by alexandercampbell-wf** [comment]
+1

@dustinhiatt-wf @beaulyddon-wf @tylertreat-wf

**Comment by dustinhiatt-wf** [comment]
+1

**Comment by tylertreat-wf** [comment]
+1

**Comment by beaulyddon-wf** [comment]
+1

**Comment by jlhonora** [comment]
Great! Awesome repo, looking forward to make some time to do more meaningful contribs.

**Comment by alexandercampbell-wf** [comment]
Thanks @jlhonora!

**Pull request merged**